### ISSUE TITLE
hadProtocol should target nsg:Protocol objects

### ISSUE THREAD

**Issue by wizmer** [INCF/neuroshapes#168]
In https://github.com/INCF/neuroshapes/blob/c35a5deccf533607b09a49d0d4e8d188c59bd0da/modules/simulation/src/main/resources/schemas/neurosciencegraph/simulation/bluepyefefeatures/v0.1.3.json#L36 the hadProtocol property targets "nsg:Entity". Shouldn't it target "nsg:Protocol" ?

**Comment by Ritikseptember2003** [comment]
@wizmer 
You are correct that the nsg:hadProtocol property, based on its description - "Experimental protocols which produced traces having extracted features (protocols.json)" — should logically target "nsg:Protocol" rather than "nsg:Entity". 

The reasons are:
- Semantic Precision
- Alignment with Description
- Consistency

Some fixes also we can do like:

Replace
{
  "path": "nsg:hadProtocol",
  "description": "Experimental protocols which produced traces having extracted features(protocols.json)",
  "class": "nsg:Entity",
  "minCount": 1,
  "maxCount": 1
}
with
{
  "path": "nsg:hadProtocol",
  "description": "Experimental protocols which produced traces having extracted features(protocols.json)",
  "class": "nsg:Protocol",
  "minCount": 1,
  "maxCount": 1
}

**Comment by DhanashreePetare** [comment]
I am interested to work on the issue. Can you please assign me the same.
